### ISSUE TITLE
Add top BCs into EDMF `boundary_state!` kernel

### ISSUE THREAD

**Issue by charleskawczynski** [CliMA/ClimateMachine.jl#1743]
### Description

@simonbyrne [discovered](https://github.com/CliMA/ClimateMachine.jl/pull/1727#discussion_r521847869) that the top BCs in the EDMF kernels:
```julia
    turbconv = m.turbconv
    N = n_updrafts(turbconv)
    up_flx = fluxᵀn.turbconv.updraft
    en_flx = fluxᵀn.turbconv.environment
    if bctype == 2 # top
        @unroll_map(N_up) do i
            up_flx[i].ρaw = -n⁻ * FT(0)
            up_flx[i].ρa = -n⁻ * FT(0)
            up_flx[i].ρaθ_liq = -n⁻ * FT(0)
            up_flx[i].ρaq_tot = -n⁻ * FT(0)
        end
        en_flx.∇tke = -n⁻ * FT(0)
        en_flx.∇e_int_cv = -n⁻ * FT(0)
        en_flx.∇q_tot_cv = -n⁻ * FT(0)
        en_flx.∇e_int_q_tot_cv = -n⁻ * FT(0)
    end
```
were never actually executed. We should add in a proper BC, and make sure that it's called.

**Comment by charleskawczynski** [comment]
Closed by #2020

**Comment by charleskawczynski** [comment]
It seemed that the function was being called, but the contents were commented out.
